{"id": "49503", "title": "Connectors do not bind to their ports in Catalina.load(), but only later in start() method.", "description": "When testing with trunk at revision 957942, it was observed that the connectors do not bind to their ports in the Catalina.load() method, but only later in the start() method. This bug presumably affects version 7.0.0 as well. In Tomcat 6, there is a separate prestartup phase when starting Tomcat, which allows Connectors to bind to their ports before other parts of Tomcat are started. This is implemented by having separate Bootstrap.load(String[]) and Bootstart.start() methods [1]. However, in the current implementation, it breaks the fix for BZ 48625. To reproduce the issue, follow these steps: 1. Start the first instance successfully.", "OB": "Upon starting Tomcat, the connector.init() call in StandardService.initInternal() does not fail even if it cannot bind to the port. The exception from connector.init() is caught in Catalina.load() and rethrown as an Error, which aborts Tomcat startup.", "EB": "Expected behavior: I would expect that in StandardService.initInternal() method, the connector.init() call fails if it cannot bind to the port. The exception from connector.init() should be caught there and if the EXITONINITFAILURE flag is set, it should be rethrown, then caught in Catalina.load() and rethrown as an Error to abort Tomcat startup.", "SR": "To reproduce the issue: 1. Add the following line to conf/catalina.properties - org.apache.catalina.startup.EXITONINITFAILURE=true 2. Start two Tomcat instances with the same configuration at the same time."}